R/guides-.R

```diff
@@ -529,11 +528,28 @@ Guides <- ggproto(
     grobs <- vector("list", length(guides))
     for (i in seq_along(grobs)) {
       grobs[[i]] <- guides[[i]]$draw(
-        theme = theme, position = as.character(positions[i]),
+        theme = theme, position = positions[i],
         direction = directions[i], params = params[[i]]
       )
+      # we'll merge inside legends with same coordinate into same guide box
+      # here, we define the groups of the inside legends
+      if (identical(positions[i], "inside")) {
+        positions[i] <- paste(
+          "inside",
+          paste(attr(.subset2(grobs, i), "inside_position"), collapse = "_"),
+          sep = "_"
+        )
+      }
     }
-    keep <- !vapply(grobs, is.zero, logical(1))
+
+    # move inside legends to the last
+    positions <- factor(positions, 
+      levels = c(.trbl, unique(positions[startsWith(positions, "inside")]))
+    )
```

Comment:
Why do these need to be last?

Reply:
I found the original use levels of `c(.trbl, "inside")`, I don't know whether we must set the factor levels

Reply:
Sorry this was just curiousity on my part, it isn't something that should be changed

Reply:
Solved.